### PR TITLE
Parse M.[]

### LINKED ISSUE #2028: M.[] fails to parse
this parses:

```reason
M.[1,2]
```

this doesn't:

```reason
M.[]
```

### PR DESCRIPTION
**Pull request by anmonteiro**
fixes #2028

### CODE REVIEW

**Comment by IwanKaramazow** [comment]
What happens in a jsx context?

File: formatTest/unit_tests/expected_output/modules.re [hunk 1]
```diff
@@ -608,3 +608,6 @@ let foo =
       module Y: Y_t with type t = a,
       module Z: Z_t with type t = a,
     ) => X.a;
+
+/* https://github.com/facebook/reason/issues/2028 */
+M.([]);
```
Comment (IwanKaramazow) [inline formatTest/unit_tests/expected_output/modules.re]: do we need the parens?

Comment (anmonteiro) [reply formatTest/unit_tests/expected_output/modules.re]: I'm not sure. I'll investigate if there are conflicts.

**Comment by anmonteiro** [comment]
Thanks for the review. I added test cases for JSX context + tweaked the printer to print without parens.

**Comment by IwanKaramazow** [comment]
Thanks for the quick PR! Looks fine by me.
@chenglou ready to merge.

**Comment by chenglou** [comment]
Thanks

**Pull request merged**